bin/gbuild

```diff
@@ -119,13 +119,17 @@ def build_one_configuration(suite, arch, build_desc, reference_datetime)
       script.puts "GBUILD_COMMON_CACHE=$HOME/cache/common"
     end
     script.puts "MAKEOPTS=(-j#{@options[:num_procs]})"
-    (ref_date, ref_time) = reference_datetime.split
-    script.puts "REFERENCE_DATETIME='#{reference_datetime}'"
-    script.puts "REFERENCE_DATE='#{ref_date}'"
-    script.puts "REFERENCE_TIME='#{ref_time}'"
     script.puts
     build_desc["remotes"].each do |remote|
       dir = sanitize(remote["dir"], remote["dir"])
+      author_date = `cd inputs/#{dir} && git log --format=@%at -1 | date +"%F %T" -u -f -`.strip
+      raise "error looking up author date in #{dir}" unless $?.exitstatus == 0
+      ref_datetime = build_desc["reference_datetime"] || author_date
+      (ref_date, ref_time) = ref_datetime.split
+      script.puts "REFERENCE_DATETIME='#{ref_datetime}'"
```

Comment:
It would be easier to understand the code if the `script.puts` lines were moved outside the loop, leaving the computation of  `ref_*` inside.